### ISSUE TITLE
Call to const trait methods in const functions no longer allowed

### ISSUE THREAD

**Issue by matthieu-m** [rust-lang/rust#114808]
### Code

I tried this code:

```rust
#![feature(const_trait_impl)]

#[const_trait]
trait Foo {
    fn foo(&self) -> i32;
}

struct X;

impl const Foo for X {
    fn foo(&self) -> i32 { 4 }
}

const fn call_foo<F>(x: &F) -> i32
where
    F: ~const Foo,
{
    x.foo()
}
```

### Version it worked on

It compiled successfully on nightly 2023-07-31, it may succeed on later versions.

### Version with regression

It failed to compile on nightly 2023-08-13 with:

```
error[[E0015]](https://doc.rust-lang.org/nightly/error_codes/E0015.html): cannot call non-const fn `<F as Foo>::foo` in constant functions
  --> src/lib.rs:18:7
   |
18 |     x.foo()
   |       ^^^^^
   |
   = note: calls in constant functions are limited to constant functions, tuple structs and tuple variants
```

**Comment by fee1-dead** [comment]
See #110395. This issue is known and will be fixed once the effects rewrite is ready. Closing this issue as duplicate.

**Comment by matthieu-m** [comment]
I saw https://github.com/rust-lang/rust/issues/114677 which was quite similar.

Unfortunately, unlike it, adding `#![feature(effects)]` does not fix the failure in this case.

I'll hold on upgrading nightly for now, good luck with the rewrite!

**Comment by fee1-dead** [comment]
Yep, this will be fixed in `feature(effects)` after #114781 is merged.

**Comment by matthieu-m** [comment]
> Yep, this will be fixed in `feature(effects)` after #114781 is merged.

@fee1-dead 

So... the above PR was merged two weeks ago, and yet as of 1.74.0-nightly (2023-08-26 69e97df5ce571a777acd) the code still fails to compile: see https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=4b29f98e3e645d4e5b51fc8ad0320901.

Should this issue be reopened?

**Comment by fee1-dead** [comment]
You are supposed to enable the effects feature
